Cargo.toml

```diff
@@ -15,41 +14,40 @@ edition = "2021"
 rust-version = "1.56"
 
 [features]
-
 default = ["compute", "image", "network", "native-tls", "object-storage"]
 compute = []
 image = []
 network = []
 native-tls = ["reqwest/default-tls", "osauth/native-tls"]
-object-storage = []
+object-storage = ["tokio", "tokio-util"]
 rustls = ["reqwest/rustls-tls", "osauth/rustls"]
 
 [dependencies]
-
-bytes = { version = "^0.5" }
+async-trait = "^0.1"
 chrono = { version = "^0.4", features = ["serde"] }
 eui48 = { version = "^1.0", features = ["disp_hexstring", "serde"] }
-fallible-iterator = "^0.2.0"
 ipnet = { version = "^2.0", features = ["serde"] }
 futures = "^0.3"
 log = "^0.4"
-osauth = { version = "^0.3.3", default-features = false }
-osproto = "^0.2.1"
+osauth = { version = "^0.5", default-features = false, features = ["stream"] }
+osproto = "^0.2.0"
 pin-project = "^1.0"
-reqwest = { version = "^0.10", default-features = false, features = ["gzip", "json", "stream"] }
+reqwest = { version = "^0.11", default-features = false, features = ["gzip", "json", "stream"] }
 serde = "^1.0"
 serde_derive = "^1.0"
 serde_json = "^1.0"
 serde_yaml = "^0.8"
-tokio = { version = "^0.2.23", features = ["io-driver", "rt-core"] }
-waiter = "^0.1"
+tokio = { version = "^1.21", optional = true }
```

Comment:
Do we need all features from tokio? I'm under the impression that it's quite large.

Reply:
While looking for what features of `tokio` was in use within the code, it turns out that it is only needed for running tests and examples, and these can rely on the `dev-dependencies` to pull it in.  

I guess I was using it for the interop between `futures::io::{AsyncRead, AsyncWrite}` and `tokio::io::{AsyncRead, AsyncWrite}` within the object storage sections during the refactors, but `tokio_util` suffices on its own through its `compat` feature and the library compiles without issue with it gone, so I'll make a commit for removing it.  

I agree with you that, if the library was to be kept around, it should have had its features more granularly enabled.